tests/test_iterables/test_fold/test_collect.py

```diff
@@ -200,3 +201,11 @@ async def test_fold_collect_future_result(subtests):
             assert await Fold.collect(
                 iterable, sequence.from_value(()),
             ) == await sequence
+
+
+def test_fold_collect_recursion_limit():
+    """Ensures that ``.collect`` method is recurion safe."""
+    limit = sys.getrecursionlimit() + 1
+    iterable = [IO(1) for _ in range(limit)]
```

Comment:
Maybe it would be better to use a generator here!
What do you think @sobolevn?

Reply:
👍

Reply:
I like that suggestion. Done.